<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<HTML>
 <HEAD>
   <TITLE> [Mapnik-users] Error in python importing mapnik
   </TITLE>
   <LINK REL="Index" HREF="http://lists.berlios.de/pipermail/mapnik-users/2008-November/index.html" >
   <LINK REL="made" HREF="mailto:mapnik-users%40lists.berlios.de?Subject=Re%3A%20%5BMapnik-users%5D%20Error%20in%20python%20importing%20mapnik&In-Reply-To=%3C78d7dd350811190727q9ae3213l43b4895ee7b789aa%40mail.gmail.com%3E">
   <META NAME="robots" CONTENT="index,nofollow">
   <style type="text/css">
       pre {
           white-space: pre-wrap;       /* css-2.1, curent FF, Opera, Safari */
           }
   </style>
   <META http-equiv="Content-Type" content="text/html; charset=us-ascii">
   <LINK REL="Previous"  HREF="001372.html">
   <LINK REL="Next"  HREF="001375.html">
 </HEAD>
 <BODY BGCOLOR="#ffffff">
   <H1>[Mapnik-users] Error in python importing mapnik</H1>
    <B>Nguyen Vu Hung</B> 
    <A HREF="mailto:mapnik-users%40lists.berlios.de?Subject=Re%3A%20%5BMapnik-users%5D%20Error%20in%20python%20importing%20mapnik&In-Reply-To=%3C78d7dd350811190727q9ae3213l43b4895ee7b789aa%40mail.gmail.com%3E"
       TITLE="[Mapnik-users] Error in python importing mapnik">vuhung16plus at gmail.com
       </A><BR>
    <I>Wed Nov 19 16:27:30 CET 2008</I>
    <P><UL>
        <LI>Previous message: <A HREF="001372.html">[Mapnik-users] Error in python importing mapnik
</A></li>
        <LI>Next message: <A HREF="001375.html">[Mapnik-users] Error in python importing mapnik
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#1374">[ date ]</a>
              <a href="thread.html#1374">[ thread ]</a>
              <a href="subject.html#1374">[ subject ]</a>
              <a href="author.html#1374">[ author ]</a>
         </LI>
       </UL>
    <HR>  
<!--beginarticle-->
<PRE>Never mind, I fixed it myself.

At some point, I've rebuilt boost but I didn't rebuild mapnik.

I clean mapnik by running

/usr/bin/python scons/scons.py  -c

and then

/usr/bin/python scons/scons.py
/usr/bin/python scons/scons.py  install

and it worked.

Btw, under CentOs 5, the libraries generated by boost have *gcc41* in
their names,
so I had to do:

cd /home/vuhung/opt/boost/lib
ln -s libboost_filesystem-gcc41-mt.solibboost_filesystem-mt.so
ln -s libboost_regex-gcc41-mt.solibboost_regex-mt.so
ln -s libboost_iostreams-gcc41-mt.solibboost_iostreams-mt.so
ln -s libboost_program_options-gcc41-mt.solibboost_program_options-mt.so
ln -s libboost_thread-gcc41-mt.solibboost_thread-mt.so
ln -s libboost_python-gcc41-mt.solibboost_python-mt.so

If you think this worths to be added to:
<A HREF="http://trac.mapnik.org/wiki/InstallationTroubleshooting">http://trac.mapnik.org/wiki/InstallationTroubleshooting</A>
I will do.

On Wed, Nov 19, 2008 at 10:39 PM, Nguyen Vu Hung &lt;<A HREF="https://lists.berlios.de/mailman/listinfo/mapnik-users">vuhung16plus at gmail.com</A>&gt; wrote:
&gt;<i> Hi all,
</I>&gt;<i>
</I>&gt;<i> I've tried to install mapnik and import it from python. Here is what I did:
</I>&gt;<i>
</I>&gt;<i> 1. CentOs 5.0, gcc 4.1.2, libstdc++-4.1.2, python  2.4.3( default).
</I>&gt;<i>
</I>&gt;<i> 2. boost_1_37_0]$ cat vh-config.sh
</I>&gt;<i> make clean
</I>&gt;<i> sh ./configure --prefix=$HOME/opt/boost --with-icu --with-python=/usr/bin/python
</I>&gt;<i> make
</I>&gt;<i> make install
</I>&gt;<i>
</I>&gt;<i> 3. mapnik-0.5.1]$ cat vh-scons.sh
</I>&gt;<i> /usr/bin/python scons/scons.py
</I>&gt;<i> BOOST_INCLUDES=$HOME/opt/boost/include/boost-1_37
</I>&gt;<i> BOOST_LIBS=$HOME/opt/boost/lib
</I>&gt;<i> sudo /usr/bin/python scons/scons.py
</I>&gt;<i> BOOST_INCLUDES=$HOME/opt/boost/include/boost-1_37
</I>&gt;<i> BOOST_LIBS=$HOME/opt/boost/lib install
</I>&gt;<i>
</I>&gt;<i> Then I tried to import mapnik, but ended up with an error.
</I>&gt;<i> 4. $ /usr/bin/python
</I>&gt;<i> Python 2.4.3 (#1, May 24 2008, 13:47:28)
</I>&gt;<i> [GCC 4.1.2 20070626 (Red Hat 4.1.2-14)] on linux2
</I>&gt;<i> Type &quot;help&quot;, &quot;copyright&quot;, &quot;credits&quot; or &quot;license&quot; for more information.
</I>&gt;&gt;&gt;&gt;<i> from mapnik import *
</I>&gt;<i> Traceback (most recent call last):
</I>&gt;<i>  File &quot;&lt;stdin&gt;&quot;, line 1, in ?
</I>&gt;<i>  File &quot;/usr/lib/python2.4/site-packages/mapnik/__init__.py&quot;, line 31, in ?
</I>&gt;<i>    from _mapnik import *
</I>&gt;<i> ImportError: /home/vuhung/opt/boost/lib/libboost_python-gcc41-mt-1_37.so.1.37.0:
</I>&gt;<i> undefined symbol: _Py_RefTotal
</I>&gt;<i>
</I>&gt;<i> I am new to both boost, python and mapnik.
</I>&gt;<i>
</I>&gt;<i> Any ideas?
</I>&gt;<i>
</I>&gt;<i>
</I>&gt;<i> --
</I>&gt;<i> Best Regards,
</I>&gt;<i> Nguyen Hung Vu ( Nguy&#7877;n V&#361; H&#432;ng )
</I>&gt;<i> vuhung16plus{remove}@gmail.dot.com , YIM: vuhung16 , Skype: vuhung16dg
</I>&gt;<i> A brief profile: <A HREF="http://www.hn.is.uec.ac.jp/~vuhung/Nguyen.Vu.Hung.html">http://www.hn.is.uec.ac.jp/~vuhung/Nguyen.Vu.Hung.html</A>
</I>&gt;<i>
</I>


-- 
Best Regards,
Nguyen Hung Vu ( Nguy&#7877;n V&#361; H&#432;ng )
vuhung16plus{remove}@gmail.dot.com , YIM: vuhung16 , Skype: vuhung16dg
A brief profile: <A HREF="http://www.hn.is.uec.ac.jp/~vuhung/Nguyen.Vu.Hung.html">http://www.hn.is.uec.ac.jp/~vuhung/Nguyen.Vu.Hung.html</A>
</PRE>

<!--endarticle-->
    <HR>
    <P><UL>
        <!--threads-->
	<LI>Previous message: <A HREF="001372.html">[Mapnik-users] Error in python importing mapnik
</A></li>
	<LI>Next message: <A HREF="001375.html">[Mapnik-users] Error in python importing mapnik
</A></li>
         <LI> <B>Messages sorted by:</B> 
              <a href="date.html#1374">[ date ]</a>
              <a href="thread.html#1374">[ thread ]</a>
              <a href="subject.html#1374">[ subject ]</a>
              <a href="author.html#1374">[ author ]</a>
         </LI>
       </UL>

<hr>
<a href="https://lists.berlios.de/mailman/listinfo/mapnik-users">More information about the Mapnik-users
mailing list</a><br>
</body></html>
